From f16f1fb9ed405908fbaef4bc1a17a1e5d8a1fbe6 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 7 Nov 2024 09:00:07 +0100
Subject: [PATCH 1349/1695] drivers: stm32_hpdma: restore Address Remapping
 configuration on resume

HPDMA Address Remapping in located in SYSCFG and this configuration is lost
during low power sequences.
Restore the Address Remapping configuration in stm32_hpdma_pm_resume() and
replace error code returns with panic(), as stm32_hpdma_pm_resume() doesn't
manage errors.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 902fdefacd816f03f3a19581e64719d620673e2a drivers: stm32_hpdma: restore Address Remapping configuration on resume]
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I6eb1ce09f4ddd9af84471c5f6639f450cd6c2a39
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/416975
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_hpdma.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/core/drivers/stm32_hpdma.c b/core/drivers/stm32_hpdma.c
index 86b8aaf4a..e804e0a9d 100644
--- a/core/drivers/stm32_hpdma.c
+++ b/core/drivers/stm32_hpdma.c
@@ -103,7 +103,7 @@ static TEE_Result stm32_hpdma_syscfg_set_arcr(struct hpdma_pdata *hpdma_d)
 	if (SYSCON_ID(SYSCON_SYSCFG, offset) != SYSCFG_HPDMAARCR) {
 		EMSG("HPDMA offset for axi address remap is incorrect: %u",
 		     offset);
-		return TEE_ERROR_BAD_PARAMETERS;
+		panic();
 	}
 
 	/* Verify that mask is consistent */
@@ -112,7 +112,7 @@ static TEE_Result stm32_hpdma_syscfg_set_arcr(struct hpdma_pdata *hpdma_d)
 	      mask == SYSCFG_HPDMAARCR_HPDMA3AREN)) {
 		EMSG("HPDMA bitmask for axi address remap is incorrect: %u",
 		     mask);
-		return TEE_ERROR_BAD_PARAMETERS;
+		panic();
 	}
 
 	stm32mp_syscfg_write(SYSCFG_HPDMAARCR, mask, mask);
@@ -330,6 +330,8 @@ static TEE_Result parse_dt(const void *fdt, int node,
 static void stm32_hpdma_pm_resume(struct hpdma_pdata *hpdma)
 {
 	apply_rif_config(hpdma, true);
+
+	stm32_hpdma_syscfg_set_arcr(hpdma);
 }
 
 static void stm32_hpdma_pm_suspend(struct hpdma_pdata *hpdma)
-- 
2.47.2

